File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        HashMap<Integer,Integer> mp=new HashMap<>();
        int n=nums.length;
        int sum=0;
        int count=0;
        for(int i=0;i<n;i++){
             sum+=nums[i];
             mp.put(sum,i);
             if(sum==k){count++;}
             else if(mp.containsKey(sum-k)){
                if(mp.get(sum-k)!=i){
                    count++;
                }
             }
        }
        return count;
        
    }
}

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution(object):
    def subarraySum(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        #nums.sort()
        l=0
        r=0
        n=len(nums)
        sum=0
        ans=0
        while(r<n):
            sum+=nums[r]
            while(l<r and (sum>k or sum<0)):
               sum-=nums[l]
               l+=1
            if(sum==k):
                ans+=1
            r+=1
        return ans
            


        